djedi/plugins/form.py

```diff
@@ -0,0 +1,79 @@
+import json
+import cio
+from djedi.plugins.base import DjediPlugin
+from django import forms
+
+
+def deprefix(s):
+    # Remove prefix (anything including and before __)
+    return s.rpartition('__')[-1]
+
+
+def get_custom_render_widget(cls):
+    class CustomRenderWidget(cls):
+        def render(self, *args, **kwargs):
+            name = kwargs.pop("name", None)
+
+            if not name:
+                name = args[0]
+                args = args[1:]
+
+            name = deprefix(name)
+
+            return super(CustomRenderWidget, self).render(
+                "data[%s]" % name,
+                *args,
+                **kwargs
+            )
+
+    return CustomRenderWidget
+
+
+class BaseEditorForm(forms.Form):
+    def __init__(self, *args, **kwargs):
+        super(BaseEditorForm, self).__init__(*args, **kwargs)
+
+        for field in list(self.fields.keys()):
+            self.fields[field].widget.__class__ = get_custom_render_widget(
+                self.fields[field].widget.__class__
+            )
+
+
+class FormsBasePlugin(DjediPlugin):
+    ext = None
+
+    @property
+    def forms(self):
+        return {}
+
+    def get_editor_context(self, **context):
+        uri = context.get('uri')
+
+        loaded = cio.load(uri)
+        data = {}
+
+        if loaded:
+            data = loaded.get('data') or {}
```

Comment:
Loading initial form data from storage here is not needed, since the client side editor already loads the data using the api, once the plugin editor is rendered.